build: intc_gic.c CPU-count assert fails on platforms whose CPU compatible has no binding

**Describe the bug**

Since commit 7afcac3f51e ("drivers: interrupt_controller: Use new CPU macro to iterate through CPUs"), building any GIC platform whose CPU nodes' compatible has no devicetree binding fails with:

```
include/zephyr/toolchain/gcc.h:87:36: error: static assertion failed: "The count of CPU Cores nodes in dts is less than CONFIG_MP_MAX_NUM_CPUS"
drivers/interrupt_controller/intc_gic.c:34:1: note: in expansion of macro 'BUILD_ASSERT'
```

Root cause: `DT_FOREACH_CPU_STATUS_OKAY_SEP()` keeps only `/cpus` children whose `device_type` property is visible to the devicetree API. Property macros are only generated for properties declared by a matched binding (`dts/bindings/cpu/cpu.yaml` declares `device_type`). A CPU node whose compatible has no binding therefore gets no `device_type` macro even though the DTS sets `device_type = "cpu"`, every CPU is filtered out, `cpu_mpid_list[]` ends up empty, and the assert (added in 2021 by 460a962b876, harmless until now) fires.

Affected in-tree platforms and their binding-less compatibles:

- `cortex_r8_virtual` (`arm,cortex-r8f`)
- `rcar_h3ulcb/r8a77951/a57` (`arm,cortex-a57`)
- `qnxhv_vm` (`arm,armv8`)

This also shows up as red twister-build CI on unrelated PRs (e.g. #112959), since CI rebases PRs onto main before building.

**To Reproduce**

On current main:

```
./scripts/twister -p cortex_r8_virtual -s sample.kernel.synchronization --build-only
```

**Expected behavior**

The build succeeds; a CPU node with `status = "okay"` and `device_type = "cpu"` in the DTS is enumerated by `DT_FOREACH_CPU_STATUS_OKAY_SEP()` regardless of whether its compatible has a binding, or the missing bindings exist.

**Impact**

Build failure on main for the three platforms above and CI noise on every PR touching nothing related.

**Environment**

- main at ce3d455bf34
- Zephyr SDK 1.0.1, arm-zephyr-eabi-gcc

Fix proposed in #114109 (add the three missing CPU bindings). Whether `DT_FOREACH_CPU_*` should be robust against binding-less CPU compatibles in general may deserve separate consideration.

--- a/dts/bindings/cpu/arm,cortex-a57.yaml
+++ b/dts/bindings/cpu/arm,cortex-a57.yaml
@@ -1,0 +1,8 @@
+# Copyright (c) 2026 BayLibre SAS
+# SPDX-License-Identifier: Apache-2.0
+
+description: ARM Cortex-A57 CPU
+
+compatible: "arm,cortex-a57"
+
+include: cpu.yaml

